summer-of-code/week-02/wk2-homework-submissions/soc_wk2_cert_cristina_tarantino.test.py

```diff
@@ -0,0 +1,542 @@
+                           'w': 0, 'x': 0, 'y': 0, 'z': 0}
+        self.assertEqual(homework.char_frequency_dict_from_alphabet(test_text), expected_result)
+
+    def test_char_frequency_dict_from_alphabet_file(self):
+        filename = "../alice_in_wonderland.txt"
+        # source http://www.gutenberg.org/files/11/11-0.txt
+
+        # open the file in read only mode
+        # source https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files
+        with open(filename) as f:
+            read_data = f.read()
+        f.closed
+
+        expected_result = {'a': 9804, 'b': 1746, 'c': 3003, 'd': 5469, 'e': 15396, 'f': 2383, 'g': 2944, 'h': 7890,
+                           'i': 8634, 'j': 235, 'k': 1290, 'l': 5211, 'm': 2466, 'n': 8053, 'o': 9480, 'p': 1968,
+                           'q': 220, 'r': 6612, 's': 7269, 't': 12204, 'u': 3979, 'v': 963, 'w': 2952, 'x': 176,
+                           'y': 2584, 'z': 80}
+
+        self.assertEqual(homework.char_frequency_dict_from_alphabet(read_data), expected_result)
+
+
+class TestCharFrequencyDictOnlyPresentAlphas(unittest.TestCase):
+    def test_char_frequency_dict_only_present_alphas_string(self):
+        test_text = "Ciao io sono Cristina"
+        expected_result = {'c': 2, 'i': 4, 'a': 2, 'o': 4, 's': 2, 'n': 2, 'r': 1, 't': 1}
+        self.assertEqual(homework.char_frequency_dict_only_present_alphas(test_text), expected_result)
+
+    def test_char_frequency_dict_only_present_alphas_file(self):
+        filename = "../alice_in_wonderland.txt"
+        # source http://www.gutenberg.org/files/11/11-0.txt
+
+        # open the file in read only mode
+        # source https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files
+        with open(filename) as f:
+            read_data = f.read()
+        f.closed
+
+        expected_result = {'a': 9804, 'b': 1746, 'c': 3003, 'd': 5469, 'e': 15396, 'f': 2383, 'g': 2944, 'h': 7890,
+                           'i': 8634, 'j': 235, 'k': 1290, 'l': 5211, 'm': 2466, 'n': 8053, 'o': 9480, 'p': 1968,
+                           'q': 220, 'r': 6612, 's': 7269, 't': 12204, 'u': 3979, 'v': 963, 'w': 2952, 'x': 176,
+                           'y': 2584, 'z': 80}
+
+        self.assertEqual(homework.char_frequency_dict_only_present_alphas(read_data), expected_result)
+
+class TestSong(unittest.TestCase):
+    @mock.patch('sys.stdout', new_callable=StringIO)
+    def test_sing_me_a_song(self, mock_stdout):
+        lyric = ["Happy birthday to you",
+                 "I don't want to get sued",
+                 "So I'll stop right there"]
+
+        expected_result = lyric[0] + "\n" + lyric[1] + "\n" + lyric[2] + "\n"
+
+        happy_bday = homework.Song(lyric)
+        happy_bday.sing_me_a_song()
+        self.assertEqual(mock_stdout.getvalue(), expected_result)
+
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
Good work on the test cases ! Really proud of what you have done so far !

Reply:
👍 Thank you